mailet/standard/src/test/java/org/apache/james/transport/mailets/StripAttachmentTest.java

```diff
@@ -155,201 +245,483 @@ public void testSimpleAttachment2() throws MessagingException, IOException {
                 .build();
-        recover.init(mci2);
 
-        Mailet onlyText = new OnlyText();
-        onlyText.init(FakeMailetConfig.builder()
+        expectedException.expect(MailetException.class);
+        expectedException.expectMessage("Could not compile regex [.****\\.tmp]");
+        mailet.init(mci);
+    }
+
+    @Test
+    public void initShouldThrowOnWrongNotPattern() throws MessagingException {
+        Mailet mailet = new StripAttachment();
+
+        FakeMailetConfig mci = FakeMailetConfig.builder()
                 .mailetName("Test")
-                .build());
+                .setProperty("directory", folderPath)
+                .setProperty("remove", "all")
+                .setProperty("notpattern", ".****\\.tmp")
+                .build();
+
+        expectedException.expect(MailetException.class);
+        expectedException.expectMessage("Could not compile regex [.****\\.tmp]");
+        mailet.init(mci);
+    }
+
+    @Test
+    public void initShouldSetRemoveAttachments() throws MessagingException {
+        StripAttachment mailet = new StripAttachment();
+
+        FakeMailetConfig mci = FakeMailetConfig.builder()
+                .mailetName("Test")
+                .setProperty("directory", folderPath)
+                .setProperty("remove", "matched")
+                .setProperty("pattern", ".*\\.tmp")
+                .build();
+        mailet.init(mci);
+        
+        assertThat(mailet.removeAttachments).isEqualTo("matched");
+    }
+
+    @Test
+    public void serviceShouldThrowWhenUnretrievableMessage() throws MessagingException {
+        StripAttachment mailet = new StripAttachment();
+        
+        Mail mail = mock(Mail.class);
+        when(mail.getMessage())
+            .thenThrow(new MessagingException("Test exception"));
+
+        expectedException.expect(MailetException.class);
+        expectedException.expectMessage("Could not retrieve message from Mail object");
+        
+        mailet.service(mail);
+    }
+
+    @Test
+    public void serviceShouldThrowWhenUnretrievableContentTypeMessage() throws MessagingException {
+        StripAttachment mailet = new StripAttachment();
+
+        MimeMessage message = mock(MimeMessage.class);
+        Mail mail = mock(Mail.class);
+        when(mail.getMessage())
+            .thenReturn(message);
+        when(message.isMimeType("multipart/*"))
+            .thenThrow(new MessagingException("Test exception"));
+
+        expectedException.expect(MailetException.class);
+        expectedException.expectMessage("Could not retrieve contenttype of MimePart.");
+        
+        mailet.service(mail);
+    }
+
+    @Test
+    public void getMailetInfoShouldReturn() throws MessagingException {
+        StripAttachment mailet = new StripAttachment();
 
+        assertThat(mailet.getMailetInfo()).isEqualTo("StripAttachment");
+    }
+
+    @Test
+    public void analyseMultipartPartMessageShouldReturnFalseWhenPartIsNotMultipart() throws Exception {
```

Comment:
This PR relies too much on visible for testing IMO.

It shows either : 
 - almost no extraction to external classes
 - a bad testing strategy trying to test too much things at once